app/backend/src/couchers/servicers/account.py

```diff
@@ -31,7 +32,7 @@ def _check_password(user, field_name, request, context):
             # wrong password
             context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.INVALID_USERNAME_OR_PASSWORD)
 
-    if request.HasField(field_name) and not user.hashed_password:
+    elif request.HasField(field_name):
```

Comment:
Leave this as `elif not user.has_password and request.HasField(field_name)`; that's more explicit and much clearer what's going on

Reply:
What if I add a comment, instead of testing the same thing twice.

Reply:
I feel like the comments here are sufficient. They kind of walk you through the possibilities.

Reply:
OK, fair enough.